osgp/protocol-adapter-mqtt/osgp-protocol-adapter-mqtt/src/main/java/org/opensmartgridplatform/adapter/protocol/mqtt/application/services/SubscriptionService.java

```diff
@@ -47,11 +51,20 @@ public SubscriptionService(
     this.mqttClientDefaults = mqttClientDefaults;
   }
 
+  public void setMqttClient(final MqttClient mqttClient) {
```

Comment:
It seems this method is only added for testing purposes? 
Have you considered removing this method here and using `ReflectionTestUtils` to set the mqttClient in the test?

Reply:
Updated the unit test.